common/djangoapps/third_party_auth/strategy.py

```diff
@@ -41,6 +42,15 @@ def setting(self, name, default=None, backend=None):
                 if error_url:
                     return error_url
 
+        # Special case: we want to get this particular setting directly from the provider database
+        # entry if possible; if we don't have the information, fall back to the default behavior.
```

Comment:
This shouldn't need to be a special case? If you look at the docstring above, this code already does the lookup you describe, first checking the relevant provider database entry, then falling back to this. I think the only "special case" here may be the capitalization. You should correct the capitalization with a two-line entry [here](https://github.com/open-craft/edx-platform/blob/3f0ae94d43812c6ad23fe1dd4cb2d4c12d22f157/common/djangoapps/third_party_auth/models.py#L306-L319), rather than in this method.